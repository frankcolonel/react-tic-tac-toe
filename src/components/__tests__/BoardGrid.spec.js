import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { initialState } from '../../reducers/game';

import BoardGrid from '../BoardGrid';

function setupComponent() {
  const props = {
    grid: initialState.boardGrid,
    handleNewMark: jest.fn(),
    isWinnerMark: jest.fn(),
    getPlayerSymbol: jest.fn(),
  };

  const wrapper = shallow(
    <BoardGrid {...props} />,
  );

  return {
    props,
    wrapper,
  };
}

describe('<BoardGrid />', () => {
  it('should render without exploding', () => {
    const { wrapper } = setupComponent();

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
