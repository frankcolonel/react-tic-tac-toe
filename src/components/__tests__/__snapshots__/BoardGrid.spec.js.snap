exports[`<BoardGrid /> should render without exploding 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "listStyleType": "none",
      "padding": 0,
      "width": 300,
    }
  }>
  <BoardSpace
    column={0}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={0}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={0}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={1}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={0}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={2}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={1}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={0}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={1}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={1}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={1}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={2}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={2}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={0}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={2}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={1}
    isWinnerMark={[Function]}
    value={null} />
  <BoardSpace
    column={2}
    getPlayerSymbol={[Function]}
    handleNewMark={[Function]}
    index={2}
    isWinnerMark={[Function]}
    value={null} />
</ul>
`;
